core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -374,6 +372,21 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
     zkClient.getClusterId.getOrElse(zkClient.createOrGetClusterId(CoreUtils.generateUuidAsBase64))
   }
 
+  private def registerBroker(zkClient: KafkaZkClient): BrokerInfo = {
+    if (!config.advertisedListenersEndPoints.isEmpty) {
```

Comment:
I wonder if we should check `config.advertisedListeners` instead? I guess you are expecting that we would fail to bind if the listener was incorrectly specified?

Reply:
I was thinking we want this check only if advertised.listeners configured.  but it makes sense to check  if listener was incorrectly specified. updated the code